test/fail_compilation/test18385.d

```diff
@@ -0,0 +1,31 @@
+/*
+REQUIRED_ARGS: -de
+TEST_OUTPUT:
+---
+fail_compilation/test18385.d(13): Deprecation: function `test18385.foo(double)` cannot be overloaded with another `extern(C)` function at fail_compilation/test18385.d(12)
+fail_compilation/test18385.d(18): Deprecation: function `test18385.S.foo(double)` cannot be overloaded with another `extern(C)` function at fail_compilation/test18385.d(17)
+---
+*/
+
+extern (C):
+
+void foo(int) { }
+void foo(double) { }
+
+struct S
+{
+    static void foo(int) {}
+    static void foo(double) {}
```

Comment:
Please remind me why this shouldn't be allowed? These two symbols are part of a struct and therefore will get mangled as D symbols, even though the linkage is `extern (C)`.

The changelog entry says: "Note that this change is only relevant for mangling schemes that do no support
 overloading". The D mangling scheme **does** support overloading. Make up your minds.

Reply:
As discussed above, the current behaviour of `extern(C)` not changing the mangling inside of a struct is a bug because the function still adheres to the C ABI.

The current implementation could cater for that special case but I don't think that it should. Having the deprecation now will allow for a smoother transition when fixing `extern(C)` inside `struct`s.